cfme/tests/automate/test_instance.py

```diff
@@ -354,3 +354,117 @@ def test_quota_source_value(request, entity, search, copy_quota_instance, generi
     provision_request.wait_for_request(method='ui')
     request.addfinalizer(lambda: provision_request.remove_request(method="rest"))
     assert result.validate(wait="60s")
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1441353])
+@pytest.mark.parametrize("process", ["miq_stop", "miq_abort"])
+def test_miq_stop_abort_with_state_machines(request, process, domain, klass, namespace):
+    """
+    Bugzilla:
+        1441353
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Automate
+        tags: automate
+    """
+    state1 = fauxfactory.gen_alpha()
+    state2 = fauxfactory.gen_alpha()
+    state3 = fauxfactory.gen_alpha()
+    for state in [state1, state2, state3]:
+        klass.schema.add_fields({'name': state, 'type': 'State'})
```

Comment:
I think it will be better if you send all the fields at once instead of sending individual fields.
Something like: `klass.schema.add_fields(*[{'name': state, 'type': 'State'} for state in [state1, state2, state3]])` or any other suitable way.